website/docs/reference/warehouse-profiles/bigquery-profile.md

```diff
@@ -215,6 +215,51 @@ Database Error in model debug_table (models/debug_table.sql)
   compiled SQL at target/run/bq_project/models/debug_table.sql
 ```
 
+### Service Account Impersonation
+You may want your models to be built using a dedicated service account that has
+elevated access to read or write data to the specified project or dataset.
+Typically, this requires you to create a service account key for running under
+development or on your CI server. By specifing the email address of the service
+account you want to build models as, you can use [Application Default Credentials](
+https://cloud.google.com/sdk/gcloud/reference/auth/application-default) or the
+service's configured service account (when running in GCP) to assume the identity
+of the service account with elevated permissions.
+
+This allows you to reap the advantages of using federated identity for developers
+(via ADC) without needing to grant individual access to read and write data
+directly, and without needing to create separate service account and keys for
+each user. It also allows you to completely eliminate the need for service
+account keys in CI as long as your CI is running on GCP (Cloud Build, Jenkins,
+GitLab/Github Runners, etc).
+
+To use this functionality, first create the service account you want to
+impersonate. Then grant users that you want to be able to impersonate
+this service account the `roles/iam.serviceAccountTokenCreator` role on
+the service account resource. Then, you also need to grant the service
+account the same role on itself. This allows it to create short-lived
+tokens identifying itself, and allows your human users (or other service
+accounts) to do the same. More information on this scenario is available
+[here](https://cloud.google.com/iam/docs/understanding-service-accounts#directly_impersonating_a_service_account).
+
+Once you've granted the appropriate permissions. You'll need to enable
+the [IAM Service Account Credentials API](https://console.cloud.google.com/apis/library/iamcredentials.googleapis.com).
+Then, just add the `impersonate_service_account` option to your BigQuery
+profile configuration:
+
+```yaml
+my-profile:
+  target: dev
+  outputs:
+    dev:
+      type: bigquery
+      method: oauth
+      project: abc-123
+      dataset: my_dataset
+      impersonate_service_account: dbt-runner@yourproject.iam.gserviceaccount.com
```

Comment:
I needed to wrap this in quotes:
```suggestion
      impersonate_service_account: "dbt-runner@yourproject.iam.gserviceaccount.com"
```

Reply:
This was copied from my working config, what sort of issue were you having without the quotes?

Reply:
When it wasn't quoted, I was getting:
```
Encountered an error:
Runtime Error
  Unable to generate access token, if you're using impersonate_service_account, make sure your initial account has the "roles/iam.serviceAccountTokenCreator" role on the account you are trying to impersonate.
```
After I quoted, it worked.

Reply:
Could you possibly try it again without quotes? I'm thinking maybe it was some eventual consistency in the IAM which can take up to 7 minutes to fully propagate worst-case. If it still doesn't work I want to investigate before merging; it should be behaving the same for us.

Reply:
You're right! I tried again without quotes and it worked, so this was just a coincidence of timing. Mind adding a note to these docs, indicating that users may need to wait up to 7 minutes? I could see that being a source of confusion